### ISSUE TITLE
Support authorization code flow without the client secret for single-page-apps

### ISSUE THREAD

**Issue by jimmyshiau** [quire-api/quire-api#8]
https://aaronparecki.com/oauth-2-simplified/#single-page-apps

**Comment by cor3000** [comment]
avoid the deprecated 'implicit flow':
https://medium.com/securing/what-is-going-on-with-oauth-2-0-and-why-you-should-not-use-it-for-authentication-5f47597b2611

**Comment by rafjaf** [comment]
Is there any plan to implement this?

**Comment by eyrewiut** [comment]
Hi, know there is probably quite a lot to work on right now, but are there any updates or a workaround for this? Would be a great help. Thanks

**Comment by rafjaf** [comment]
@eyrewiut In the meanwhile I plan to locate the authentication function (including the secret and client id) in something like pipedream.com, which will be called by xhr and return the token. That's the only way presently to preserve the secret.

**Comment by rafjaf** [comment]
@eyrewiut In the meanwhile I plan to implement the authentication function in something like pipedream.com, which will be called by the single page app through xhr and return the token. That's the only way that I can presently see to preserve the secret.

**Comment by eyrewiut** [comment]
Hi @rafjaf, Thanks, that helps. Getting a little bit off topic, but how do you plan on returning the token, just as json result? I don't know much about auth so this helps 😄

**Comment by rafjaf** [comment]
@eyrewiut Yes indeed, as a json result. This is perhaps not ideal but better than nothing given the lack of implementation of a solution so far by Quire developers...

**Comment by nicolalopatriello** [comment]
So, this is the same problem of which i trying to do? 

In practice: there is a way to consume Quire api from a server application with, for example, an **api-key** instead of oauth2 workflow?
If yes, where i can find it? If no, there is a way or workaround for this?

I need to create a simple client app **without** authentication with Quire side.

Thanks

**Comment by tomyeh** [comment]
PKCE supported. Refer to [Change log](https://github.com/quire-api/quire-api/blob/master/CHANGES.md#apr-13-2026).
